engine/src/externalv1.cpp

```diff
@@ -2046,6 +2157,44 @@ static MCExternalError MCExternalInterfaceQuery(MCExternalInterfaceQueryTag op,
 	return kMCExternalErrorNone;
 }
 
+
+////////////////////////////////////////////////////////////////////////////////
+
+MCExternalError MCExternalLicenseCheckEdition(unsigned int p_options, unsigned int p_min_edition)
+{
+	unsigned int t_current_edition;
+	switch(MClicenseparameters . license_class)
+	{
+		case kMCLicenseClassNone:
+			t_current_edition = kMCExternalLicenseTypeNone;
+			break;
+			
+		case kMCLicenseClassCommunity:
+			t_current_edition = kMCExternalLicenseTypeCommunity;
+			break;
+			
+		case kMCLicenseClassCommercial:
+			t_current_edition = kMCExternalLicenseTypeIndy;
+			break;
+			
+		case kMCLicenseClassProfessional:
+			t_current_edition = kMCExternalLicenseTypeBusiness;
+			break;
+			
+		default:
+			abort();
+			break;
+	}
+	
+	if (t_current_edition < p_min_edition)
+	{
+		s_current_external -> SetWasLicensed(false);
```

Comment:
It might be worth at least adding an assertion that s_current_external is not nil here